core/src/test/java/io/grpc/CallOptionsTest.java

```diff
@@ -70,86 +67,103 @@
   }
 
   @Test
   public void noStrayModifications() {
-    assertTrue(equal(allSet,
-        allSet.withAuthority("blah").withAuthority(sampleAuthority)));
-    assertTrue(equal(allSet,
-        allSet.withDeadline(Deadline.after(314, NANOSECONDS))
-            .withDeadline(sampleDeadline)));
-    assertTrue(equal(allSet,
-          allSet.withAffinity(Attributes.EMPTY).withAffinity(sampleAffinity)));
+    assertThat(equal(allSet, allSet.withAuthority("blah").withAuthority(sampleAuthority)))
+        .isTrue();
+    assertThat(
+        equal(allSet,
+            allSet.withDeadline(Deadline.after(314, NANOSECONDS)).withDeadline(sampleDeadline)))
+        .isTrue();
+    assertThat(
+        equal(allSet,
+            allSet.withAffinity(Attributes.EMPTY).withAffinity(sampleAffinity)))
+        .isTrue();
   }
 
   @Test
   public void mutation() {
     Deadline deadline = Deadline.after(10, SECONDS);
     CallOptions options1 = CallOptions.DEFAULT.withDeadline(deadline);
-    assertNull(CallOptions.DEFAULT.getDeadline());
-    assertEquals(deadline, options1.getDeadline());
+    assertThat(CallOptions.DEFAULT.getDeadline()).isNull();
+    assertThat(deadline).isSameAs(options1.getDeadline());
+
     CallOptions options2 = options1.withDeadline(null);
-    assertEquals(deadline, options1.getDeadline());
-    assertNull(options2.getDeadline());
+    assertThat(deadline).isSameAs(options1.getDeadline());
+    assertThat(options2.getDeadline()).isNull();
   }
 
   @Test
   public void mutateExecutor() {
     Executor executor = MoreExecutors.directExecutor();
     CallOptions options1 = CallOptions.DEFAULT.withExecutor(executor);
-    assertNull(CallOptions.DEFAULT.getExecutor());
-    assertSame(executor, options1.getExecutor());
+    assertThat(CallOptions.DEFAULT.getExecutor()).isNull();
+    assertThat(executor).isSameAs(options1.getExecutor());
+
     CallOptions options2 = options1.withExecutor(null);
-    assertSame(executor, options1.getExecutor());
-    assertNull(options2.getExecutor());
+    assertThat(executor).isSameAs(options1.getExecutor());
+    assertThat(options2.getExecutor()).isNull();
   }
 
   @Test
   public void withDeadlineAfter() {
-    Deadline deadline = CallOptions.DEFAULT.withDeadlineAfter(1, MINUTES).getDeadline();
+    Deadline actual = CallOptions.DEFAULT.withDeadlineAfter(1, MINUTES).getDeadline();
     Deadline expected = Deadline.after(1, MINUTES);
-    assertDeadlineEquals(deadline, expected, 10, MILLISECONDS);
+
+    assertAbout(deadline()).that(actual).isWithin(10, MILLISECONDS).of(expected);
+  }
+
+  @Test
+  public void toStringMatches_noDeadline_default() {
+    String expected = "CallOptions{deadline=null, authority=authority, affinity={sample=blah}, "
+        + "executor=class io.grpc.internal.SerializingExecutor, compressorName=null}";
+    String actual = allSet
+        .withDeadline(null)
+        .withExecutor(new SerializingExecutor(directExecutor()))
```

Comment:
Keep `directExecutor()` consistent. There's the other usage in `mutateExecutor()`.